lms/security.py

```diff
@@ -101,6 +142,37 @@ def forget(self, request):
         pass
 
 
+class LTILaunchesSecurityPolicy(LTIUserSecurityPolicy):
+    """
+    Security policy for LTI launches
+    """
+
+    @classmethod
+    def authenticated_userid(cls, request):
+        try:
+            return get_userid_from_lti_user(cls._lti_user(request))
+        except ValidationError:
+            None
+
+    def identity(self, request):
+        try:
+            lti_user = self._lti_user(request)
+        except ValidationError as err:
+            return Identity("", [], err)
+
+        return get_identity_from_lti_user(lti_user)
+
+    @staticmethod
+    @lru_cache(maxsize=1)
+    def _lti_user(request):
```

Comment:
Need to rethink this method, both 

authenticated_userid
identity

need to execute this at the moment when validation fails.